src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,581 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
+        {
+            if (IndexWordsFloorNumber.Contains(RightIndex(rrIndex)))
             {
-                if (houseNumber != null && houseNumber.Contains("-") && Int32.TryParse(houseNumber.Split('-')[0].Trim(), out hnr))
+                // verdiepnummer
+                return new List<HouseNumberWithSubaddress>
+                {
+                    new HouseNumberWithSubaddress(
+                        RemovePrecedingZeros(houseNumber),
+                        null,
+                        $"{RemovePrecedingZeros(LeftIndex(rrIndex))}.0")
+                };
+            }
+
+            // geen verdiepnummer
+            return new List<HouseNumberWithSubaddress>
+            {
+                new HouseNumberWithSubaddress(
+                    $"{RemovePrecedingZeros(houseNumber)}_{RemovePrecedingZeros(LeftIndex(rrIndex))}",
+                    RightIndex(rrIndex),
+                    null)
+            };
+        }
+
+        private static List<HouseNumberWithSubaddress> MatchHouseNumberRanges(string? streetName, string? houseNumber, string rrindex0)
+        {
+            string? houseNumber1 = null;
+            string? houseNumber2 = null;
+
+            if (rrindex0 == "0000" || IndexWordsWithoutBisNumber.Contains(rrindex0))
+            {
+                if (houseNumber != null && houseNumber.Contains("-") && int.TryParse(houseNumber.Split('-')[0].Trim(), out _))
                 {
                     houseNumber1 = houseNumber.Split('-')[0].Trim();
                     houseNumber2 = houseNumber.Substring(houseNumber.IndexOf('-') + 1).Trim();
                 }
-                if (houseNumber != null && houseNumber.Contains("/") && Int32.TryParse(houseNumber.Split('/')[0].Trim(), out hnr))
+
+                if (houseNumber != null && houseNumber.Contains("/") && int.TryParse(houseNumber.Split('/')[0].Trim(), out _))
                 {
                     houseNumber1 = houseNumber.Split('/')[0].Trim();
                     houseNumber2 = houseNumber.Substring(houseNumber.IndexOf('/') + 1).Trim();
                 }
-                if (houseNumber != null && houseNumber.Contains("+") && Int32.TryParse(houseNumber.Split('+')[0].Trim(), out hnr))
+
+                if (houseNumber != null && houseNumber.Contains("+") && int.TryParse(houseNumber.Split('+')[0].Trim(), out _))
                 {
                     houseNumber1 = houseNumber.Split('+')[0].Trim();
                     houseNumber2 = houseNumber.Substring(houseNumber.IndexOf('+') + 1).Trim();
                 }
-                if (houseNumber != null && new Regex("[0-9] [0-9]").IsMatch(houseNumber) && Int32.TryParse(houseNumber.Split(' ')[0].Trim(), out hnr))
+
+                // TODO: [0-9]+ [0-9]+ ?
+                if (houseNumber != null && new Regex("[0-9] [0-9]").IsMatch(houseNumber) &&
```

Comment:
Shouldn't this be `[0-9]+ [0-9]+` instead?

Reply:
In retrospect, this will match `XX XXX` so `blablaXX XXXblalbla` is seen as a match too :)

It would still be cleaner with `[0-9]+ [0-9]+` explicitly.